### ISSUE TITLE
Models posterior to epoch 1 not working

### ISSUE THREAD

**Issue by Kailegh** [SeanNaren/deepspeech.pytorch#411]
Hi, I have tried to train a few models and I am facing some problems. After each epoch model continues to get better. However, when I try to load a model beyond epoch 1 from disk results are totally wrong:

> --model-path models/deepspeech_checkpoint_epoch_1_iter_27000.pth
> Test Summary Average WER 64.740 Average CER: 48.125
> --model-path models/deepspeech_checkpoint_epoch_2_iter_27000.pth
> Test Summary Average WER 613.68 Average CER: 197.308
> --model-path models/deepspeech_1.pth.tar
> Test Summary Average WER 613.68 Average CER: 197.308
> --model-path models/deepspeech_2.pth.tar
> Test Summary Average WER 613.68 Average CER: 197.308
> --model-path models/deepspeech_final.pth
> Test Summary Average WER 613.68 Average CER: 197.308

All the models beyond epoch one have no sense and the same results, do any of you know why this might be happening?  
Thanks a lot!

**Comment by BuaaAlban** [comment]
excuse me ? have you trained with the default parameters?

**Comment by Kailegh** [comment]
I use this to start a training:
`python train.oy --train-manifest ../data/train.csv --val-manifest ../data/dev.csv --sample-rate 8000 --batch-size 8 --cuda --tensorboard --checkpoint --checkpoint-per-batch 3000 --num-workers 8 --lr 0.001 --epochs 5 --silent`

Next results are from the experiment I ran yesterday:
```
Training Summary Epoch: [1] Time taken(s) 21916 Average Loss 36.055
Validation Summary Epoch: [1] Average WER 100.000 Average CER 84.182

Training Summary Epoch: [2] Time taken(s) 21753 Average Loss 19.499
Validation Summary Epoch: [2] Average WER 100.000 Average CER 84.182

```
During training I received I single warning that is the following:
```
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
```

If I use the test script with the last checkpoint of epoch 1:
```
python test.py --model-path models/deepspeech_chechpoint_epoch_1_iter_45000.pth --test-manifest ../data/test.csv --cuda --num-workers 8
Test Summary Average WER 41.516 Average CER 10.693
```

same with model epoch 2:
```
python test.py --model-path models/deepspeech_chechpoint_epoch_2_iter_45000.pth --test-manifest ../data/test.csv --cuda --num-workers 8
Test Summary Average WER 100.000 Average CER 84.023
```

I have no clue why this might be happening, I would appreciate any help on this  
Thanks a lot

**Comment by BuaaAlban** [comment]
Maybe you need to set your lr  smaller

**Comment by Kailegh** [comment]
I do not think it is that, if I train again from this model using 
`--from-model models/deepspeech_chechpoint_epoch_1_iter_45000.pth --finetune
`
it gets better on the validation test until it completes an epoch, the it gives 100% WER again, it is like it gets broken when completing an epoch

**Comment by BuaaAlban** [comment]
on the contrary, I got a ovious loss drop  when  it completes an epoch at the training stage, you should just try one epoch or two for a little lr 0.0003. My loss stables at 100-200 at epoch 1  and below 100 at epoch 2. if it doesn't work , you may consider other problems


朱博康
北京航空航天大学
中法工程师学院
地址：北京市海淀区学院路37号
邮编：100191
电话：13051576002
邮箱：buaazbk@buaa.edu.cn


Zhu bokang
Beihang University
Ecole Centrele de Pekin
Adress:No.37 College Road, Haidian District, Beijing
Tel: 13051576002
E-mail: buaazbk@buaa.edu.cn
On 5/29/2019 18:12，Kailegh<notifications@github.com> wrote：

I do not think it is that, if I train again from this model using
--from-model models/deepspeech_chechpoint_epoch_1_iter_45000.pth --finetune
it gets better on the validation test until it completes an epoch, the it gives 100% WER again, it is like it gets broken when completing an epoch

**Comment by Kailegh** [comment]
I have tried reducing the lr to 0.0001 and the problem is the same, when it finishes an epoch the model gets broken.

I think this is an important problem and would appreciate the help of anybody, @SeanNaren  do you have any idea?

**Comment by SeanNaren** [comment]
definitely looks like your model diverges, I assume the datasets you're using are not publically available to have a look at? Try swapping the rnns to LSTMs (--rnn-type lstm) which may cause some out of memory issues, and might need you to reduce the hidden-size.

LSTMs tend to be better for stability, but looking forward to here if this helps!

**Comment by Kailegh** [comment]
I have just tried 600 neurons and using lstm cells and the same problem occurs again.  
The validation WER is 100 and CER 84.182. Which is exactly the same as with the models before. Howhever the last checkpoint of epoch one achieves results that indeed have sense, just in my first posts.   
I think the model must be getting corrupted at some point at the end of the first epoch. Am I the only person who has this error?  
The data comes from audiobooks, voxforge... but I have trained models using mozilla DeepSpeech and Kaldi and models are trained without any error.
